File: app/src/androidTest/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/TestApplication.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate;

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.app.AppComponent;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.app.AppModule;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.components.MockAppComponent;

public class TestApplication extends App {

    private MockAppComponent mockAppComponent;

    public MockAppComponent getMockAppComponent() {
        return (MockAppComponent) getAppComponent();
    }

    @Override
    public AppComponent getAppComponent() {
        if (mockAppComponent == null) {
            mockAppComponent = DaggerMockAppComponent.builder()
                    .appModule(new AppModule(this))
                    .build();
        }
        return mockAppComponent;
    }
}

File: app/src/androidTest/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/TestApplication.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.app.AppComponent
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.app.AppModule
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.components.MockAppComponent

class TestApplication : App() {

    private lateinit var mockAppComponent: MockAppComponent

    override var appComponent: AppComponent
        get() {
            if (mockAppComponent == null) {
                @Suppress("Annotator")
                mockAppComponent = DaggerMockAppComponent.builder()
                        .appModule(AppModule(this))
                        .build()
            }
            return mockAppComponent
        }
        set(value) {
            super.appComponent = value
        }
}
